index.js

```diff
@@ -10,9 +10,85 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+
+
+const fetchColors = ({ name, hex, compName, compHex }) => 
+  
+  {
+   return fetch(COLORS)
+
+    .then((response) => {
+      if (!response.ok) {
+       throw new Error(`HTTP error, status = ${response.status}`);
+      }
+      return response.json();
+    })
+
+    .then((data) => {
+
+      //create an empty array for appending future results, use "let" because it's block-scoped
+      let result = []
+
+      //iterate through data by each element
+      data.forEach(element => {
+
+        //since Complimentary colors and hex codes are by themselves arrays, 
+        //we first make them their own total element "comp" and then iterate through it later
+        //therefore our "element" only has 3 arguments at the moment
+        const {name: colorName, hex: colorHex, comp} = element;
+
+          //check if the name in the JSON value includes the passed name
+          if (name                                                    //checks it is defined first
+            && colorName.toLowerCase().includes(name.toLowerCase())   //case insenstitive 
```

Comment:
Outsource this comparison to a function. If the requirement changed to be case-sensitive you would have to change the comparison in multiple places.